implementation/jwt-auth/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java

```diff
@@ -836,10 +846,11 @@ Optional<JWTAuthContextInfo> getOptionalContextInfo() {
         contextInfo.setTokenAge(mpJwtVerifyTokenAge.orElse(null));
         contextInfo.setJwksRefreshInterval(jwksRefreshInterval);
         contextInfo.setForcedJwksRefreshInterval(forcedJwksRefreshInterval);
+        contextInfo.setJwksRetainOnErrorDuration(jwksRetainOnErrorDuration);
         Set<SignatureAlgorithm> resolvedAlgorithm = mpJwtPublicKeyAlgorithm;
         if (signatureAlgorithm.isPresent()) {
             if (signatureAlgorithm.get().getAlgorithm().startsWith("HS")) {
-                if (!NONE.equals(resolvedVerifyKeyLocation) && resolvedVerifyKeyLocation == mpJwtLocation) {
+                if (!NONE.equals(resolvedVerifyKeyLocation) && resolvedVerifyKeyLocation.equals(mpJwtLocation)) {
```

Comment:
I know this looks suspicious and as far as I recall it was already discussed, but it is actually correct. In any case, let's deal with unrelated improvements in another PR